sharding/nodesSetup_test.go

```diff
@@ -651,7 +651,7 @@ func TestNodesSetup_IfNodesWithinMaxShardLimitEquivalentDistribution(t *testing.
 
 	ns = createAndAssignNodes(*ns, 2169)
 
-	ns2 := &(*ns)
+	ns2 := ns
```

Comment:
please revert it as now the test is working properly.

Previously the test was doing the following:
1. creating creating a node setup object, storing it's reference in ns
2. filling this node setup with dummy created nodes and assigning them to shards, according with it's configured max num shards 100
3. creating a clone of this nodes setup object and assigning its reference to n2
4. changing the n2 genesis max num shards to 3 and creating again dummy nodes (using the same model as before) and assigning them to shards with respect to the new maximum number of shards
5. verifying that in the second case (ns2) we reach the exact same configuration of nodes as in ns because although even if ns had a different configuration for max num shards then ns2, both configs only created enough nodes for 3 shards which was within the limit of both. This test was verifying the backwards compatibility of adding a maximum number of shards constraint in the nodes setup.

now with this change, ns2 directly points to the same object as ns, so the test will no longer properly check the backwards compatibility of this change.